src/lib/plane.js

```diff
@@ -1,94 +1,174 @@
+           Math.abs(t2 - p2) <= e * Math.max(1.0, Math.abs(t2), Math.abs(p2)) &&
+           Math.abs(t3 - p3) <= e * Math.max(1.0, Math.abs(t3), Math.abs(p3));
+  }
+
+  /**
+   * Check for exact equality against the given plane.
+   *
+   * @param {Plane} p Plane to compare
+   * @returns {Boolean} Equality
+   */
+  equals(p) {
+    return this[0] === p[0] &&
+           this[1] === p[1] &&
+           this[2] === p[2] &&
+           this[3] === p[3];
+  }
+
+  /**
+   * Execute the provided function once for each plane in the given array.
+   *
+   * @param {(Float32Array|Array)} arr Array to traverse
+   * @param {Function} cb Callback
+   * @returns {void}
+   */
+  static forEach(arr, cb) {
+    if (arr.length % LENGTH !== 0) {
+      throw new Error('Invalid length');
+    }
+
+    const l = arr.length;
+    const p = new Plane();
+
+    for (let i = 0; i < l; i += LENGTH) {
+      p[0] = arr[i];
+      p[1] = arr[i + 1];
+      p[2] = arr[i + 2];
+      p[3] = arr[i + 3];
+
+      cb.call(this, p, i / LENGTH, arr);
+
+      arr[i]     = p[0];
+      arr[i + 1] = p[1];
+      arr[i + 2] = p[2];
+      arr[i + 3] = p[3];
+    }
+  }
+
+  /**
+   * Generate an iterator capable of iterating over an array as plane indices.
+   *
+   * @param {(Float32Array|Array)} arr Array to iterate
+   * @returns {Iterator} New iterator
+   */
+  static *keys(arr) {
+    if (arr.length % LENGTH !== 0) {
+      throw new Error('Invalid length');
+    }
+
+    const l = arr.length / LENGTH;
+
+    for (let i = 0; i < l; i++) {
+      yield i;
+    }
+  }
+
+  /**
+   * Create a new plane with a variable number of arguments.
+   *
+   * @param {...*} args Arguments for new plane
+   * @returns {Plane} New plane
+   */
+  static of(...args) {
+    if (args.length === 0) {
+      return new Plane();
+    } else {
+      return new Plane(args);
+    }
+  }
+
+  /**
+   * Generate an iterator capable of iterating over an array as planes.
+   *
+   * @param {(Float32Array|Array)} arr Array to iterate
+   * @returns {Iterator} New iterator
+   */
+  static *values(arr) {
+    if (arr.length % LENGTH !== 0) {
+      throw new Error('Invalid length');
+    }
+
+    const l = arr.length;
+    const p = new Plane();
+
+    for (let i = 0; i < l; i += LENGTH) {
+      p[0] = arr[i];
+      p[1] = arr[i + 1];
+      p[2] = arr[i + 2];
+      p[3] = arr[i + 3];
+
+      yield p;
+
+      arr[i]     = p[0];
+      arr[i + 1] = p[1];
+      arr[i + 2] = p[2];
+      arr[i + 3] = p[3];
+    }
+  }
 }
 
-/**
- * Create new plane from given values.
- *
- * @param {Number} nx The plane normal's x coord
- * @param {Number} ny The plane normal's y coord
- * @param {Number} nz The plane normal's z coord
- * @param {Number} c The plane's constant
- *
- * @returns {Plane} The new plane
- */
-export function fromValues(nx, ny, nz, c) {
-
-  const plane = create();
+Plane.DEFAULT = DEFAULT;
 
-  // Normal
-  plane[0] = nx;
-  plane[1] = ny;
-  plane[2] = nz;
+Plane.LENGTH = LENGTH;
```

Comment:
Guess we're staying as vanilla as possible here, and not use `static` from https://github.com/tc39/proposal-class-fields?

Reply:
I'd like to avoid adding class fields until it reaches stage 4 (ie sees a browser implementation), I think. That proposal has undergone quite a bit of discussion over the years, and I'd be frustrated to see it change just after we start banking on it heavily.

Reply:
Sounds good! 👍